utils/table_to_annotations.py

```diff
@@ -0,0 +1,277 @@
+    match Biospecimen metadata with column names to create key:value pairs,
+    apply annotations to each file,
+    return Individual Key:File Id and Model Key: File Id dictionaries
+    """
+    
+    individual_tup_list = []
+    model_tup_list = []
+    
+    component, table_id, column_list = specimen_info_tuple
+    data_table = get_annotations_table(syn, table_id)
+    count = 0
+    for _, row in data_table.iterrows():
+        id = row["Biospecimen_id"]
+        if id in file_tuples.keys():
+            file_id = file_tuples[id]
+            individual_id = row["Individual Key"]
+            model_id = row["Model Key"]
+            individual_tup = (individual_id, file_id)
+            individual_tup_list.append(individual_tup)
+            model_tup = (model_id, file_id)
+            model_tup_list.append(model_tup)
+            biospecimen_annotations = list(zip(column_list, list(row)))
+            biospecimen_annotations = apply_annotations_to_entity(syn, component, file_id, biospecimen_annotations)
+            count += 1
+    
+    print(f"Biospecimen annotations applied to {count} entities")
+    return dict(individual_tup_list), dict(model_tup_list)
+
+
+def collect_record_annotations(syn, info_tuple: tuple[str, str, str], tuple_dict: dict):
+    """Collect all entries from a Synapse table,
+    select entries where primary key (e.g. Individual_id) matches foreign key (e.g. Individual Key),
+    based on dictionary output from collect_biospecimen_annotations,
+    apply annotations to each file"""
+    
+    component, table_id, column_list = info_tuple
+    data_table = get_annotations_table(syn, table_id)
+    count = 0
+    for _, row in data_table.iterrows():
+        id = row[f"{component}_id"]
+        if id in tuple_dict.keys():
+            file_id = tuple_dict[id]
+            annotations = list(zip(column_list, list(row)))
+            annotations = apply_annotations_to_entity(syn, component, file_id, annotations)
+            count += 1
+    
+    print(f"{component} annotations applied to {count} entities")
+
+
+def apply_annotations_to_entity(syn, component: str, entity_id: str, new_annotations: list[tuple[str, str]]):
+    """Apply annotations to a Synapse entity by:
+    retrieving current annotations,
+    converting new_annotations tuple to key:value pairs within the retrieved annotation object,
+    storing modified annotation object in Synapse.
+    Note that only keys with non-empty values will be applied."""
+    
+    entity_annotations = syn.get_annotations(entity_id)
+    filtered_annotations = [tup for tup in new_annotations if len(tup[1]) > 0]
+    for annot in filtered_annotations:
+        entity_annotations[annot[0].replace(" ", "")] = annot[1]
+    entity = syn.set_annotations(entity_annotations)
+    print(f"\n{component} annotations applied to Synapse entity: {entity_id}")
+
+
+def main():
+
+    syn = (
+        synapseclient.login()
+    ) 
```

Comment:
nit: line of code is short, so no need to break it over multiple lines.

```suggestion
    syn = synapseclient.login()
```